runtime/lua/vim/lsp/callbacks.lua

```diff
@@ -198,12 +198,12 @@ end
 
 -- Add boilerplate error validation and logging for all of these.
 for k, fn in pairs(M) do
-  M[k] = function(err, method, params, client_id)
-    local _ = log.debug() and log.debug('default_callback', method, { params = params, client_id = client_id, err = err })
+  M[k] = function(err, method, params, client_id, bufnr)
+    local _ = log.debug() and log.debug('default_callback', method, { params = params, client_id = client_id, err = err, bufnr = bufnr })
```

Comment:
Can you drop the first part of this? Just call `log.debug(...)` directly. Idk why we recommended to people the other part.

Reply:
I think it's to avoid executing logic as part of the second `log.debug(..)` call. In this case it would be the table creation with all the parameters. Not sure if it's worth avoiding that in lua.

I think if the decision is to avoid this style of logging then it would also make sense to remove the possibility to use the single args variant for a quick log-level check - otherwise this would be prune for personal-preference bike-shedding.